compiler/noirc_frontend/src/lexer/lexer.rs

```diff
@@ -303,9 +303,12 @@ impl<'a> Lexer<'a> {
         self.next_char();
 
         let contents_start = self.position + 1;
-
-        let word = self.eat_while(None, |ch| ch != ']');
-
+        let mut attribute_name = self.eat_while(None, |ch| ch != '(' && ch != ']');
+        if Self::is_fv_attribute(&attribute_name) {
+            return Ok(Self::into_fv_attribute_token(&attribute_name).unwrap().into_span(start, self.position))
+        } 
+        attribute_name.push_str(&self.eat_while(None, |ch| ch != ']'));
+        let word = attribute_name;
```

Comment:
Now that I think of it, this is pretty neat, actually!

Even though the default lexer just puts everything in a word, like a dummy, you manage to stop the lexing process and then proceed with the lexing behavior as if the contents are normal Noir code.

Good job.